src/datadog/grok/matchers/date.rs

```diff
@@ -256,37 +268,45 @@ pub fn apply_date_filter(value: &Value, filter: &DateFilter) -> Result<Value, Gr
         datetime.push_str("+0000");
     };
 
-    if let Some(tz) = filter
-        .regex
-        .captures(&original_value)
-        .and_then(|caps| caps.name("tz"))
-    {
-        let tz = tz.as_str();
-        let tz: Tz = tz.parse().map_err(|error| {
-            warn!(message = "Error parsing tz", tz = %tz, % error);
-            GrokRuntimeError::FailedToApplyFilter(filter.to_string(), original_value.to_string())
-        })?;
-        replace_sec_fraction_with_dot(filter, &mut datetime);
-        let naive_date = NaiveDateTime::parse_from_str(&datetime, &strp_format).map_err(|error|
+    if filter.with_tz_capture {
+        if let Some(tz) = filter
+            .regex
+            .captures(&original_value)
+            .and_then(|caps| caps.name("tz"))
         {
```

Comment:
Could you use an `expect` here, since the `else` branch is just a `panic!`?
```suggestion
        let tz = filter
            .regex
            .captures(&original_value)
            .and_then(|caps| caps.name("tz"))
            .expect("Filter should contain tz capture");
```